sklearn/compose/_column_transformer.py

```diff
@@ -473,6 +525,72 @@ def transform(self, X):
 
         return _hstack(list(Xs), self.sparse_output_)
 
+    def inverse_transform(self, X):
+        """Apply inverse transform
+
+        All estimators in ``transformers`` must support ``inverse_transform``.
+
+        Parameters
+        ----------
+        X: array-like, shape = [n_samples, n_transformed_features]
+            Data samples, where ``n_samples`` is the number of samples and
+            ``n_features`` is the number of features.
+
+        Returns
+        -------
+        Xt : array-like or DataFrame of shape [n_samples, n_features]
+
+        """
+        check_is_fitted(self, 'transformers_')
+
+        if self._invert_error:
+            raise ValueError("Unable to invert: {}".format(self._invert_error))
+
+        X_subs = (_get_column(X, key) for key in self._output_indices)
+        inv_transformers = []
+        get_weight = (self.transformer_weights or {}).get
+
+        for (name, trans, column), sub in zip(self.transformers_, X_subs):
+            if trans == 'passthrough':
+                trans = FunctionTransformer(
+                    validate=False, accept_sparse=True, check_inverse=False)
+
+            inv_transformers.append((name, trans, sub, get_weight(name)))
+
+        Xs = Parallel(n_jobs=self.n_jobs)(
+            delayed(_inverse_transform_one)(
+                trans, X_sel, weight)
+            for _, trans, X_sel, weight in inv_transformers)
+
+        if not Xs:
+            # All transformers are None
+            return np.zeros((X.shape[0], 0))
+
+        if self._X_is_sparse:
+            inverse_Xs = sparse.lil_matrix((Xs[0].shape[0],
+                                            self._n_features_in))
+        else:
+            inverse_Xs = np.zeros((Xs[0].shape[0], self._n_features_in))
+        for indices, inverse_X in zip(self._input_indices, Xs):
+            if sparse.issparse(inverse_X):
+                if self._X_is_sparse:
+                    inverse_Xs[:, indices] = inverse_X
+                else:
+                    inverse_Xs[:, indices] = inverse_X.toarray()
+            else:
+                if inverse_X.ndim == 1:
```

Comment:
is this possible?

Reply:
Are you referring to `inverse_x.ndim == 1` or `inverse_Xs[:, indices] = ...`?

Reply:
I meant the second about `inverse_x` being 1-dimensional. I would think that any valid sklearn transformer should always return 2D output? (but maybe I am overlooking something). And so I would expect input for `inverse_transform` to have the same constraint.

Reply:
`preprocessing.LabelEncoder`'s `transform` and `inverse_transform` returns 1-D arrays.

Reply:
LabelEncoder shouldn't be used on `X`

Reply:
This PR was updated to remove the check for 1D outputs.